### PR TITLE
Fix impression reserve logic

### PR DESCRIPTION
**Pull request by solocommand**
This broke in the final iteration of #5, since `NaN` is not equivalent to `null`.

- Move impression reserve logic into a function to allow testing
- Update impression reserve logic to check for `NaN`
- Add unit tests for each condition (no settings, account-only, placement of `0` and placement of non-zero)

### CODE REVIEW

File: src/services/campaign-delivery.js [hunk 1]
```diff
@@ -111,15 +111,28 @@ module.exports = {
...
+   * @param placement The placement model
+   * @param account The account model
+   */
+  calculateImpressionReserve({
+    placement,
+    account,
+  }) {
+    const rp = parseInt(placement.get('reservePct'), 10);
+    const ap = account.get('settings.reservePct');
+    return (Number.isInteger(rp) ? rp : (ap || 0)) / 100;
```
Comment (zarathustra323) [inline src/services/campaign-delivery.js]: A float would also be acceptable

**Review by zarathustra323 — CHANGES_REQUESTED** [review]

File: src/services/campaign-delivery.js (hunk 1 above)
Comment (zarathustra323) [inline src/services/campaign-delivery.js]: Couldn't it just be if rp != null && !Number.isNaN

Comment (solocommand) [reply src/services/campaign-delivery.js]: The GraphQL definitions for the [account](https://github.com/parameter1/fortnight-graph/blob/master/src/graph/definitions/account.graphql#L24) and [placement](https://github.com/parameter1/fortnight-graph/blob/master/src/graph/definitions/placement.graphql#L26) are both `Int` types, so I don't think floats _should_ be accepted. I could swap this to use `Number.isFinite` if we want to allow it, though.

Comment (zarathustra323) [reply src/services/campaign-delivery.js]: The GraphQL definitions have no bearing on this function. Just do the isNaN check and be done with it.

Comment (zarathustra323) [reply src/services/campaign-delivery.js]: Or wrap it with a parseInt

**Review by zarathustra323 — CHANGES_REQUESTED** [review]

File: src/services/campaign-delivery.js [hunk 2]
```diff
@@ -111,15 +111,28 @@ module.exports = {
...
+   * @param placement The placement model
+   * @param account The account model
+   */
+  calculateImpressionReserve({
+    placement,
+    account,
+  }) {
+    const rp = parseInt(placement.get('reservePct'), 10);
+    const ap = account.get('settings.reservePct');
+    return (rp !== null && !Number.isNaN(rp) ? rp : (ap || 0)) / 100;
```
Comment (zarathustra323) [inline src/services/campaign-delivery.js:126]: you'll want `!=` instead of `!==`

**Review by zarathustra323 — CHANGES_REQUESTED** [review]

**Review by zarathustra323 — APPROVED** [review]
